### PR TITLE
Python basics exercise

### PR DESCRIPTION
**Pull request by leemc-data-ed**
Part 4, last one! One thing to note is the the link to the Colab notebook will need to be updated right before merge, but I left in on the branch so that can be checked over as well. 

https://liascript.github.io/course/?https://raw.githubusercontent.com/arcus/education_modules/python_basics_exercise/python_basics_exercise/python_basics_exercise.md#6

### CODE REVIEW

File: python_basics_exercise/python_basics_exercise.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+import: https://raw.githubusercontent.com/arcus/education_modules/main/_module_templates/macros.md
+-->
+
+# Python Basics: Exercise
+
+@overview
+
+## Lesson Preparation
+
+For this module, you'll need access to Python, either on your own computer or in the cloud. For details about how to download Python or use Google Colab (a cloud-based notebook environment), you can take a look at the ["Accessing Python" section of our Demystifying Python module](https://liascript.github.io/course/?https://raw.githubusercontent.com/arcus/education_modules/main/demystifying_python/demystifying_python.md#9).
```
Comment (drelliche) [inline python_basics_exercise/python_basics_exercise.md:47]:
```suggestion
For this module, **you'll need access to Python, either on your own computer or in the cloud**. For details about how to download Python or use Google Colab (a cloud-based notebook environment), you can take a look at the ["Accessing Python" section of our Demystifying Python module](https://liascript.github.io/course/?https://raw.githubusercontent.com/arcus/education_modules/main/demystifying_python/demystifying_python.md#9). 
```

Just to emphasize this a bit more :)

**Review by drelliche — COMMENTED** [review]
Two quick comments, it is looking great!

File: python_basics_exercise/python_basics_exercise.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+estimated_time_in_minutes: 30
+
+@pre_reqs
+Learners should be familiar with using functions, methods, variables, lists, dictionaries, loops, and conditional statements in Python. These skills are presented in the Python Basics sequence of modules ([Functions, Methods, and Variables](https://liascript.github.io/course/?https://raw.githubusercontent.com/arcus/education_modules/main/python_basics_variables_functions_methods/python_basics_variables_functions_methods.md#1), [Lists and Dictionaries](https://liascript.github.io/course/?https://raw.githubusercontent.com/arcus/education_modules/main/python_basics_lists_dictionaries/python_basics_lists_dictionaries.md#1), and [Loops and Conditional Statements](https://liascript.github.io/course/?https://raw.githubusercontent.com/arcus/education_modules/main/python_basics_loops_conditionals/python_basics_loops_conditionals.md#1)).
+@end
+
+@learning_objectives  
+After completion of this module, learners will be able to:
+
+- Loop through a dictionary and conditionally perform an iterative task based on the values in the dictionary.
```
[the remark this answers is not in the document]
Comment (leemc-data-ed) [reply python_basics_exercise/python_basics_exercise.md:28]: Oooh yeah great idea!

**Comment by leemc-data-ed** [comment]
Edits made! How's it look now?

File: python_basics_exercise/python_basics_exercise.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+@end
+
+@learning_objectives  
+After completion of this module, learners will be able to:
+
+- Run their own Python code, either on their own computer or in the cloud.
+- Loop through a dictionary and conditionally perform an iterative task based on the values in the dictionary. 
+
+@end
+
```
Comment (drelliche) [inline python_basics_exercise/python_basics_exercise.md:32]:
```suggestion
good_first_module: false
sequence_name: python_basics
previous_sequential_module: python_basics_loops_conditionals
coding_required: true
coding_level: basic
coding_language: python

@sets_you_up_for

@end

@depends_on_knowledge_available_in

- demystifying_python
- python_basics_variables_functions_methods
- python_basics_lists_dictionaries
- python_basics_loops_conditionals

@end
```

Comment (leemc-data-ed) [reply python_basics_exercise/python_basics_exercise.md:32]: Oh yeah good catch! That's all stuff we should add to the template when we have time (not now lol)

**Review by drelliche — APPROVED** [review]

**Pull request merged**